components/brave_shields/core/browser/ad_block_default_resource_provider.cc

```diff
@@ -36,37 +37,32 @@ AdBlockDefaultResourceProvider::AdBlockDefaultResourceProvider(
 AdBlockDefaultResourceProvider::~AdBlockDefaultResourceProvider() = default;
 
 base::FilePath AdBlockDefaultResourceProvider::GetResourcesPath() {
-  if (component_path_.empty()) {
-    // Since we know it's empty return it as is.
-    return component_path_;
+  if (!component_accessor_) {
+    return base::FilePath();
   }
 
-  return component_path_.AppendASCII(kAdBlockResourcesFilename);
+  return component_accessor_->GetComponentRoot().AppendASCII(
+      kAdBlockResourcesFilename);
 }
 
 void AdBlockDefaultResourceProvider::OnComponentReady(
-    const base::FilePath& path) {
-  component_path_ = path;
-  base::FilePath resources_path = GetResourcesPath();
-
-  if (resources_path.empty()) {
+    scoped_refptr<brave_component_updater::ComponentContentsAccessor>
+        accessor) {
+  component_accessor_ = std::move(accessor);
+  if (!component_accessor_) {
```

Comment:
This seems to be an unreachable branch, because this value will never be nullptr.